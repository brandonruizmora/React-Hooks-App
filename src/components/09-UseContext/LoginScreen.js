import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

    const { setUser } = useContext(UserContext);

    return (
        <div className="container">
            <h1>LoginScreen</h1>
            <hr />
            <button 
            className="btn btn-primary"
            onClick={() => {
                setUser({
                    id: 123,
                    name: 'Brandon',
                    email: 'brandon_ruizmor@outlook.es'
                });
            }}
            >
                Login
                </button>
        </div>
    )
}
