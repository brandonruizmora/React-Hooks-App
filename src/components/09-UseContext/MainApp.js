import React from 'react'

export const MainApp = () => {
    return (
        <div className="container">
            <h1>MainApp</h1>
            <hr />
        </div>
    )
}
