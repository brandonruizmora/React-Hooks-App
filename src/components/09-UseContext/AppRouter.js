import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route
  } from 'react-router-dom';
  import { HomeScreen } from './HomeScreen';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div className="container">
                <Switch>
                    <Route path="/" exact component={HomeScreen} />
                    <Route path="/about" exact component={AboutScreen} />
                    <Route path="/login" exact component={LoginScreen} />
                </Switch>
            </div>
        </Router>
    )
}
