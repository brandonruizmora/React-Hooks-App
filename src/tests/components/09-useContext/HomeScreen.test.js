import '../../../setupTest';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-UseContext/HomeScreen';
import { UserContext } from '../../../components/09-UseContext/UserContext';

describe('Pruebas en <HomeScreen />', () => {

    const user = {
        id: 1,
        nombre: 'Brandon',
        email: 'brandon@gmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen />
        </UserContext.Provider>
    );

    test('Debe de retornar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
        
    });
    


});
