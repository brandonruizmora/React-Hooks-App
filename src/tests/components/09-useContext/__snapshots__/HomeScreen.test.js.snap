// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> Debe de retornar el componente correctamente 1`] = `
<div
  className="container"
>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
    "id": 1,
    "nombre": "Brandon",
    "email": "brandon@gmail.com"
}
  </pre>
</div>
`;
